spec/BowlingSpec.js

```diff
@@ -0,0 +1,186 @@
+    });
+
+  });
+
+  describe('Final Scores', function() {
+    it('can end the game when there have been 10 frames', function() {
+      for (i = 1; i < 21; i++) {
+        bowling.bowl(3);
+      };
+      expect( function(){ bowling.bowl(1); } ).toThrow(new Error('it is the end of the game'));
+    });
+
+    it('can play a game without any spares or strikes', function() {
+      for (i = 1; i < 21; i++) {
+        bowling.bowl(4);
+      };
+      expect( function(){ bowling.bowl(1); } ).toThrow(new Error('it is the end of the game'));
+      expect(bowling.finalScore()).toEqual(80);
+    });
+
+    it('can play a game with a strike and a spare, non-sequential', function() {
+      bowling.bowl(10);
+      for (i = 1; i < 15; i++) {
+        bowling.bowl(4);
+      };
+      bowling.bowl(5);
+      bowling.bowl(5);
+      bowling.bowl(4);
+      bowling.bowl(4);
+      expect( function(){ bowling.bowl(1); } ).toThrow(new Error('it is the end of the game'));
+      expect(bowling.finalScore()).toEqual(96)
+    });
+
+    it('can play a game with all spares except for last round', function() {
+      for(i = 1; i < 19; i++) {
+        bowling.bowl(5);
+      };
+      bowling.bowl(4)
+      bowling.bowl(4)
+      expect( function(){ bowling.bowl(1); } ).toThrow(new Error('it is the end of the game'));
+      expect(bowling.finalScore()).toEqual(142)
+    });
+
+    it('can play a game with all spares', function() {
+      for (i = 1; i < 21; i++) {
+        bowling.bowl(5);
+      };
+      bowling.bowl(5);
+      expect( function(){ bowling.bowl(1); } ).toThrow(new Error('it is the end of the game'));
+      expect(bowling.finalScore()).toEqual(150);
+    });
+
+    it('can play a game with one strike in final', function() {
+      for (i = 1; i < 11; i++) {
+        bowling.bowl(10);
+      };
+      bowling.bowl(4);
+      bowling.bowl(4);
+      expect( function(){ bowling.bowl(1); } ).toThrow(new Error('it is the end of the game'));
+      expect(bowling.finalScore()).toEqual(282);
+    });
+
+    it('can play a game with two strikes in final', function() {
+      for (i = 1; i < 12; i++) {
+        bowling.bowl(10);
+      };
+      bowling.bowl(4);
+      expect( function(){ bowling.bowl(1); } ).toThrow(new Error('it is the end of the game'));
+      expect(bowling.finalScore()).toEqual(294);
+    });
+
+    it('can play a game with a strike and a spare in final', function() {
+      for (i = 1; i < 11; i++) {
+        bowling.bowl(10);
+      };
+      bowling.bowl(5);
+      bowling.bowl(5);
+      expect( function(){ bowling.bowl(1); } ).toThrow(new Error('it is the end of the game'));
+      expect(bowling.finalScore()).toEqual(285);
+    });
+
+    it('can play a game with all strikes', function() {
+      for (i = 1; i < 13; i++) {
+        bowling.bowl(10);
+      };
+      expect( function(){ bowling.bowl(1); } ).toThrow(new Error('it is the end of the game'));
+      expect(bowling.finalScore()).toEqual(300);
+    });
+
```

Comment:
Extract a multiple roll function here - you can see that it's screaming out to be made.

Don't test for the error case every time - test it once, assume that it's gonna be working still, if you find an edge case then test that